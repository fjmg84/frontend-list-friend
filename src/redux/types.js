export const TYPES = {
  LOAD: "load/friends",
  ERROR_ACTIVE: "error/active",
};
