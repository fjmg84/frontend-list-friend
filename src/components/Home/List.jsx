import React from 'react'
import PropTypes from 'prop-types'
import { useNavigate } from 'react-router-dom'
import { avatar } from '../../images/index'

export const List = ({ friends }) => {
	const navigate = useNavigate()

	return (
		<ul>
			{friends.map((friend) => {
				return (
					<li key={friend.id}>
						<div className="card">
							<div className="card-img">
								<img src={avatar} alt={friend.img} className="small" />
								<span
									className={`availability eclipse ${
										friend.available ? 'available' : 'not-available'
									}`}
								></span>
							</div>

							<div className="card-header">
								<div className="card-header-name">
									<p className="details-name">{`${friend.first_name} ${friend.last_name}`}</p>
								</div>
								<div className="card-header-status">
									<span className="details-status">{friend.status}</span>
								</div>
							</div>

							<div className="card-actions">
								<button
									className="details-button"
									onClick={() => navigate(`./show/${friend.id}`)}
									disabled={!friend.available}
								>
									Details
								</button>
							</div>
						</div>
					</li>
				)
			})}
		</ul>
	)
}

List.protoTypes = {
	friends: PropTypes.array,
}

List.propTypes = {
	friends: PropTypes.array.isRequired,
}
