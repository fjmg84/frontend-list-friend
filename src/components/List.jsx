import React from "react";
import { useSelector } from "react-redux";
import { useNavigate } from "react-router-dom";
import image from "../photo.jpg";

export const List = () => {
  const friends = useSelector((state) => state.friendsReducer) || [];
  const navigate = useNavigate();

  return (
    <ul>
      {friends.map((friend) => {
        return (
          <li key={friend.id}>
            <div className="row">
              <div className="card">
                <img
                  src={image}
                  alt={friend.img}
                  className="rectangle1 avatar"
                />
                <span
                  className={`availability eclipse ${
                    friend.available ? "available" : "not-available"
                  }`}
                ></span>
                <div className="frame">
                  <div className="name-friend">
                    <span>{`${friend.first_name} ${friend.last_name}`}</span>
                  </div>
                  <div className="status-friend">
                    <span>{friend.status}</span>
                  </div>
                </div>
                {friend.available ? (
                  <button onClick={() => navigate(`./show/${friend.id}`)}>
                    Details
                  </button>
                ) : (
                  <button disabled>Details</button>
                )}
              </div>
            </div>
          </li>
        );
      })}
    </ul>
  );
};
