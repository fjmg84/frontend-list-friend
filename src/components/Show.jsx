import React, { useEffect, useState } from "react";
import { useParams, useNavigate } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { findOneFriend } from "../redux/actions/friends";
import { setAlert } from "../redux/actions/alerts";
import img from "../photo.jpg";

export const Show = () => {
  let { id } = useParams();
  const dispatch = useDispatch();
  const { status, msg } = useSelector((state) => state.alertsReducer);
  const friends = useSelector((state) => state.friendsReducer);
  const [statusOfTheFriend, setStatusOfTheFriend] = useState(() => {
    return friends?.filter((f) => f.id === +id)[0].status;
  });

  const [view, setView] = useState(true);
  const [friend, setFriend] = useState({});
  const bio = "";
  const navigate = useNavigate();

  useEffect(() => {
    dispatch(setAlert(true, "loading..."));
    findOneFriend(id)
      .then((friend) => {
        setFriend(friend);
        dispatch(setAlert(false, undefined));
      })
      .catch((err) => {
        console.error(err);
        dispatch(setAlert(true, "Up!!! Sorry. Not Found data of this friend."));
      });
  }, [id, dispatch]);

  const handleView = () => {
    setView(!view);
  };

  const info = () => {
    return (
      <>
        <div className="article col">
          <p className="title">Bio:</p>
          <div className="text">{friend.bio}</div>
        </div>
        <div className="divider"></div>

        <div className="article row">
          <p className="title">Phone:</p>
          <div className="">{friend.phone}</div>
        </div>
        <div className="divider"></div>

        <div>
          <div className="article row">
            <p className="title">Address</p>
            <div>{friend.address_1}</div>
          </div>

          <div className="article row">
            <p className="title">City</p>
            <div>{friend.city}</div>
          </div>

          <div className="article row">
            <p className="title">Zipcode</p>
            <div>{friend.zipcode}</div>
          </div>

          <div className="article row">
            <p className="title">State</p>
            <div>{friend.state}</div>
          </div>
        </div>
      </>
    );
  };

  const photos = () => {
    return (
      <div className="article">
        <div className="gallery">
          <img src={img} />
          <img src={img} />
          <img src={img} />

          <img src={img} />
          <img src={img} />
          <img src={img} />

          <img src={img} />
          <img src={img} />
          <img src={img} />
        </div>
      </div>
    );
  };

  const handleNavegation = () => {
    navigate(-1);
  };

  return (
    <div className="box">
      <div className="arrow">
        <span className="fill vector" onClick={() => handleNavegation()}></span>
      </div>

      <div className="body">
        <div className="header">
          <h1>Friend</h1>
          {status && <div className="error">{msg}</div>}
        </div>

        <div className="main">
          <div className="show-card">
            <div className="show-card-img">
              <img src={img} alt={friend.img} className="medium" />
              <span className="availability eclipse available"></span>
            </div>
            <div className="show-card-content">
              <p>{`${friend.first_name} ${friend.last_name}`}</p>
              <div className="show-card-header-status">
                <span>{statusOfTheFriend}</span>
              </div>

              <div className="show-card-box-data">
                <div className="nav">
                  <div className={`tab1 ${view ? "active" : "inactive"}`}>
                    <span onClick={() => handleView()}>Info</span>
                    <span className={`${view ? "line" : ""}`}></span>
                  </div>
                  <div className={`tab2 ${!view ? "active" : "inactive"}`}>
                    <span onClick={() => handleView()}>Photos</span>
                    <span className={`${!view ? "line" : ""}`}></span>
                  </div>
                </div>

                <div className="section">{view ? info() : photos()}</div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
