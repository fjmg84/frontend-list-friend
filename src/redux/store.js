import { createStore, combineReducers } from "redux";
import { friendsReducer } from "../reducer/friends";
import { errorsReducer } from "../reducer/errors";

const rootReducers = combineReducers({
  friendsReducer,
  errorsReducer,
});

export const store = createStore(
  rootReducers,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);
