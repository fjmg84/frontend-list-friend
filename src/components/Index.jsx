import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { getFriends } from "../redux/actions/friends";
import { setAlert } from "../redux/actions/alerts";
import { TYPES } from "../redux/types";
import { List } from "./List";

export const Index = () => {
  const dispatch = useDispatch();
  const { status, msg, classAlert } = useSelector(
    (state) => state.alertsReducer
  );

  useEffect(() => {
    dispatch(setAlert(true, "Loading friends...", "loading"));
    getFriends()
      .then((friends) => {
        dispatch({ type: TYPES.LOAD, payload: friends });
        dispatch(setAlert(false, undefined));
      })
      .catch((err) => {
        console.log(err);
        dispatch(setAlert(true, "Up!! Sorry. Not loading Friends :(", "error"));
      });
  }, [dispatch]);

  return (
    <div className="box">
      <div className="body">
        <div className="header">
          <h1>Friends</h1>
          {status && <div className={classAlert}>{msg}</div>}
        </div>

        <div className="main">
          <List />
        </div>
      </div>
    </div>
  );
};
