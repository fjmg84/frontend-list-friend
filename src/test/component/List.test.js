import React from "react";
import { List } from "../../components/List";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import { store } from "../../redux/store";
import { TYPES } from "../../redux/types";

describe("testing component <List/>", () => {
  const wrapper = mount(
    <Provider store={store}>
      <List />
    </Provider>
  );

  test("snapshot component <List/>", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("exists data list of user in component <List/>", async () => {
    const state = [
      {
        id: 1,
        img: "https://s3.amazonaws.com/uifaces/faces/twitter/vladarbatov/128.jpg",
        first_name: "Jeremy",
        last_name: "Davis",
        status: "At work...",
        available: true,
      },
    ];

    store.dispatch({
      type: TYPES.LOAD,
      payload: state,
    });

    const wrapper = mount(
      <Provider store={store}>
        <List />
      </Provider>
    );

    const btn = wrapper.find("li").at(0);
    expect(btn.exists()).toEqual(true);
  });
});
//https://www.figma.com/file/Pe9VVOGr2sOvfOcnzzAbvX/Dev-Test?node-id=1%3A2
