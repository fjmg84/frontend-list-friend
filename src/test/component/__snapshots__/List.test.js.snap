// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing component <List/> snapshot component <List/> 1`] = `<ul />`;
